XLS-0073-disllow-frozen-trustline-receiving/README.md

```diff
@@ -0,0 +1,92 @@
+
+##### 2.1.2.1. Order Book
+Currently, consumption of offers is allowed even if the buy amount (`TakerPays`) has been individually frozen. This proposal introduces a new change: __if the offer owner has been individually frozen on the trustline of `TakerPays` currency, the offer is considered to be _unfunded_ and therefore the step fails .__
+##### 2.1.2.1.1. Example
+Let's take a look at an example where offer is involved in transfering the funds and compare how the result differs between the current and proposed behaviors.
+
+
+```
+                                                                     Bob's Offer
++-------+         USD            +---------+        USD           +--------------+          XRP            +-------+
+| Alice | ---------------------> | Gateway | -------------------> | Sell: 100 XRP| ----------------------> | Carol |
++-------+     SendMax: 100 USD   +---------+                      +--------------+                         +-------+
+              Amount: 100 XRP                                     | Buy: 100 USD |       
+                                                                  +--------------+
+                                                                   (Bob is frozen)                             
+```
+
+Consider the following scenario:
+
+1. Issuer account issues USD to both Alice and Bob, each of which has a trustline.
+2. Issuer individually freezes Bob's trustline.
+3. Alice attempts to send XRP to Carol through a `Payment` transaction using Bob's offer to exchange USD for XRP. In the current behavior, Alice will be able to successfully send 100 XRP to Carol, and Bob receives 100 USD.
+
+This proposal suggests changing the behavior at  __Step 3__ to 
+
+> Alice attempts to send XRP to Carol through a `Payment` transaction using Bob's offer to exchange USD for XRP. Alice fails to send XRP to Carol because Bob's offer is unfunded due to the frozen trustline.
+
+### 2.2. `OfferCreate` transaction
+This proposal introduces a new change to the `OfferCreate` transaction. Currently, if the holder has been frozen, they are still allowed to submit a `OfferCreate` transaction that specifies `TakerPays` as the frozen token, allowing them to receive more funds despite already being frozen by the issuer. We propose the following change:
+* `OfferCreate` returns `tecFROZEN` if the currency of `TakerPays` has been frozen by the issuer (either individually or globally)
+
+Moreover, any existing offers with `TakerPays` currency frozen _can no longer be consumed and will be considered as an _unfunded_ offer that will be implicitly cancelled by new Offers that cross it._
+
+## 3. Invariants
+One potential invariant could be disallowing any increase in the balance of a frozen trustline as a result of a `Payment` transaction.
```

Comment:
Shouldn't this be true for _any_ transaction (and definitely including `OfferCreate`)?

Reply:
You are right. Initially I was thinking there would be exceptions.